sztp-agent/pkg/dhcp/network_manager.go

```diff
@@ -0,0 +1,69 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+Copyright (C) 2022-2023 Intel Corporation
+Copyright (c) 2022 Dell Inc, or its subsidiaries.
+Copyright (C) 2022 Red Hat.
+*/
+
+// Package dhcp implements the DHCP client
+package dhcp
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/godbus/dbus/v5"
+)
+
+// getBootstrapURLViaNetworkManager returns the sztp redirect URL via NetworkManager
+func getBootstrapURLViaNetworkManager() ([]string, error) {
+	conn, err := dbus.SystemBus()
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to system bus: %v", err)
+	}
+
+	nm := conn.Object("org.freedesktop.NetworkManager", "/org/freedesktop/NetworkManager")
+
+	var activeConnections []dbus.ObjectPath
+	err = nm.Call("org.freedesktop.DBus.Properties.Get", 0, "org.freedesktop.NetworkManager", "ActiveConnections").Store(&activeConnections)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ActiveConnections property: %v", err)
+	}
+
+	if len(activeConnections) == 0 {
+		return nil, fmt.Errorf("no active connections found")
+	}
+
+	var sztpRedirectURLs []string
+	for _, activeConnPath := range activeConnections {
+		connActive := conn.Object("org.freedesktop.NetworkManager", activeConnPath)
+
+		var dhcpPath dbus.ObjectPath
+		err = connActive.Call("org.freedesktop.DBus.Properties.Get", 0, "org.freedesktop.NetworkManager.Connection.Active", "Dhcp4Config").Store(&dhcpPath)
+		if err != nil {
+			log.Println("[INFO] failed to get Dhcp4Config property: ", err)
+			continue
+		}
+
+		dhcp := conn.Object("org.freedesktop.NetworkManager", dhcpPath)
+		var options map[string]dbus.Variant
+		err = dhcp.Call("org.freedesktop.DBus.Properties.Get", 0, "org.freedesktop.NetworkManager.DHCP4Config", "Options").Store(&options)
+		if err != nil {
+			log.Println("[INFO] failed to get Options property in DHCP4Config ", dhcpPath, ": ", err)
+			continue
+		}
+
+		if variant, ok := options["sztp_redirect_urls"]; ok {
```

Comment:
didn't you define a constant for this string earlier ?
`const sztpRedirectUrls = "sztp-redirect-urls"` ?